Enable merge queues on the mbedtls repo

bensze01 opened the pull request:
This PR contains all the work needed to enable using merge queues on the mbedtls repo.

Closes #105 

CI runs:
 - PR-head good:
   - [Internal CI][1]
   - [Open CI][7]
 - PR-merge good:
   - [Internal CI][2]
   - [Open CI][8]
 - PR-head failure:
   - [Internal CI][3]
   - [Open CI][9]
 - ABI-check failure:
   - [Internal CI][4]
   - [Open CI][10]
 - Merge queue good:
   - [Internal CI][5]
   - [Open CI][11]
 - Merge queue failure:
   - [Internal CI][6]
   - [Open CI][12]
   
[1]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-head/71/
[2]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-merge/122/
[3]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-759-head/21/
[4]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-758-merge/14/
[5]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/job/gh-readonly-queue%2Fdev%2Fbensze01%2Fmerge-queue-test%2Fpr-7958-9a478637e46ced3160db752a64573b52e66075a4/2/
[6]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/job/gh-readonly-queue%2Fdev%2Fbensze01%2Fmerge-queue-test%2Fpr-7959-07c2bea18c0277686dc42d9d88ff0b6798700f29/1/
[7]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-head/5/
[8]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-merge/4/
[9]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-759-head/2/
[10]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-758-merge/4/
[11]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/job/gh-readonly-queue%252Fdev%252Fbensze01%252Fmerge-queue-test%252Fpr-7958-9a478637e46ced3160db752a64573b52e66075a4/1/
[12]: https://ci.trustedfirmware.org/view/Mbed-TLS/job/mbed-tls-restricted-pr-ci-testing/job/gh-readonly-queue%252Fdev%252Fbensze01%252Fmerge-queue-test%252Fpr-7959-07c2bea18c0277686dc42d9d88ff0b6798700f29/1/

gilles-peskine-arm commented on vars/common.groovy:
```diff
@@ -283,14 +283,14 @@ def get_supported_windows_builds() {
  *              truncated to 140 characters (GitHub limitation).
  */
-def maybe_notify_github(context, state, description) {
+void maybe_notify_github(String context, String state, String description) {
```
Please change the parameter order to `(state, description, context)` so that the first two parameters always have the same meaning.

And I think this would be easier to understand if it was expressed with a single definition where `context` is an optional parameter.

bensze01 commented:
All the CI runs passed successfully, except [Merge queue good -Internal CI](https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/job/gh-readonly-queue%2Fdev%2Fbensze01%2Fmerge-queue-test%2Fpr-7958-9a478637e46ced3160db752a64573b52e66075a4/2/), which failed on windows due to the known flakyness of the production OpenCI jenkins master's windows executors.

gilles-peskine-arm reviewed (changes requested):
* Code review: looks correct to me, but I'd like one readability improvement.
* CI results: look good to me.
* Reporting: I think the reporting on https://github.com/Mbed-TLS/mbedtls/pull/7958 and https://github.com/Mbed-TLS/mbedtls/pull/7959 makes sense. It's hard to be sure because they have a past history, but if there are lingering problems I'm happy to resolve them as they arise.

bensze01 commented:
I think I've addressed the review comments.
Sanity check CI runs for the maybe_notify_github changes:
- Merge queue failure:
  - [Internal CI][1]
  - [Open CI][2]
  
[1]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/job/gh-readonly-queue%252Fdev%252Fbensze01%252Fmerge-queue-test%252Fpr-7959-07c2bea18c0277686dc42d9d88ff0b6798700f29/2/
[2]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/gh-readonly-queue%252Fdev%252Fbensze01%252Fmerge-queue-test%252Fpr-7959-07c2bea18c0277686dc42d9d88ff0b6798700f29/1/

gilles-peskine-arm reviewed (approved):
LGTM

paul-elliott-arm reviewed (approved):
LGTM
